app/services/optout/OptOutProposition.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.optout
+
+import models.incomeSourceDetails.TaxYear
+import models.itsaStatus.ITSAStatus
+
+case class OptOutProposition(previousTaxYear: PreviousOptOutTaxYear,
+                             currentTaxYear: CurrentOptOutTaxYear,
+                             nextTaxYear: NextOptOutTaxYear) {
+
+  private val optOutYears: Seq[OptOutTaxYear] = Seq[OptOutTaxYear](
+    previousTaxYear,
+    currentTaxYear,
+    nextTaxYear)
+
+  def availableOptOutYears: Seq[OptOutTaxYear] = optOutYears.filter(_.canOptOut)
```

Comment:
should fix: can we make it lazy val and not a function

Reply:
done